test-utils/src/main/kotlin/org/cqfn/save/testutils/MockWebServerUtils.kt

```diff
@@ -0,0 +1,137 @@
+@file:Suppress("FILE_NAME_MATCH_CLASS", "MatchingDeclarationName")
+
+package org.cqfn.save.testutils
+
+import okhttp3.mockwebserver.*
+import org.junit.Assert.assertTrue
+import org.slf4j.Logger
+import java.net.HttpURLConnection
+import java.nio.charset.Charset
+import java.util.concurrent.*
+
+private typealias ResponsesMap = ConcurrentMap<String, BlockingQueue<MockResponse>>
+
+/**
+ * Queue dispatcher with additional logging
+ *
+ * @param logger
+ */
+class LoggingQueueDispatcher(private val logger: Logger) : Dispatcher() {
+    private val responses: ResponsesMap = ConcurrentHashMap()
+    private val defaultResponses: ConcurrentMap<String, MockResponse> = ConcurrentHashMap()
+    private var failFastResponse: MockResponse = MockResponse().setResponseCode(HttpURLConnection.HTTP_NOT_FOUND)
+
+    private fun getMethodPath(fullPath: String?) = fullPath?.let { Regex("/[\\w]*").find(it)?.value } ?: ""
```

Comment:
Why only letters and only first part or URL? Is it intentional (e.g. `/api/method/1` -> `api`)?

Reply:
As far as I saw, we do not have anything like `/api/method/1 -> api` yet. But it would be better to use `Regex("^/[^?]*[^/]")`. Thanks